src/input/type_record.cc

```diff
@@ -375,6 +408,22 @@ Record::RecordData::RecordData(const string & type_name_in, const string & descr
 }
 
 
+Record::RecordData::RecordData(const RecordData &other)
```

Comment:
Such a copy constructor should be created automatically by the compiler. Try it.